_test/test_sqw/test_PixelData_operations.m

```diff
@@ -87,31 +95,172 @@ function test_compute_bin_data_correct_output_5_pages_mex_1_thread(obj)
+        iter = 0;
+        while true
+            start_idx = (iter*npix_in_page) + 1;
+            end_idx = min(start_idx + npix_in_page - 1, pix.num_pixels);
+
+            original_signal = data(obj.SIGNAL_IDX, start_idx:end_idx);
+            original_variance = data(obj.VARIANCE_IDX, start_idx:end_idx);
+
+            expected_data = data;
+            expected_data(obj.SIGNAL_IDX, start_idx:end_idx) = ...
+                    cos(original_signal);
+            expected_data(obj.VARIANCE_IDX, start_idx:end_idx) = ...
+                    abs(1 - pix.signal.^2).*original_variance;
+
+            assertEqual(pix.data, expected_data(:, start_idx:end_idx), '', ...
+                        obj.FLOAT_TOLERANCE);
+
+            if pix.has_more()
+                pix = pix.advance();
+                iter = iter + 1;
+            else
+                break;
+            end
+        end
+    end
 
-        assertEqual(s, obj.ref_s_data, '', 4.75e-4);
-        assertEqual(e, obj.ref_e_data, '', 4.75e-4);
+    function test_paged_data_returns_same_unary_op_result_as_all_in_memory(obj)
+        % the unary operation and the range the data it acts on should take
+        unary_ops = {
+            @acos, [0, 1], ...
+            @acosh, [1, 3], ...
+            @acot, [0, 1], ...
+            @acoth, [10, 15], ...
+            @acsc, [1, 3], ...
+            @acsch, [1, 3], ...
+            @asec, [1.5, 3], ...
+            @asech, [0, 1], ...
+            @asin, [0, 1], ...
+            @asinh, [1, 3], ...
+            @atan, [0, 1], ...
+            @atanh, [0, 0.5], ...
+            @cos, [0, 1], ...
+            @cosh, [0, 1], ...
+            @cot, [0, 1], ...
+            @coth, [1.5, 3], ...
+            @csc, [0.5, 2.5], ...
+            @csch, [1, 3], ...
+            @exp, [0, 1], ...
+            @log, [1, 3], ...
+            @log10, [1, 3], ...
+            @sec, [2, 4], ...
+            @sech, [0, 1.4], ...
+            @sin, [0, 3], ...
+            @sinh, [0, 3], ...
+            @sqrt, [0, 3], ...
+            @tan, [0, 1], ...
+            @tanh [0, 3], ...
+        };
+
+        % For each unary operator, perform the operation on some file-backed
+        % data and compare the result to the same operation used on the same
+        % data all held in memory
+        num_pix = 7;
+        npix_in_page = 3;
+        for i = 1:numel(unary_ops)/2
+            unary_op = unary_ops{2*i - 1};
+            data_range = unary_ops{2*i};
+
+            data = obj.get_random_data_in_range(obj.NUM_COLS_IN_PIX_BLOCK, ...
+                                                num_pix, data_range);
+            pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+            pix = pix.do_unary_op(unary_op);
+
+            transformed_data = obj.concatenate_pixel_pages(pix);
+
+            pix_in_mem = PixelData(data);
+            pix_in_mem = pix_in_mem.do_unary_op(unary_op);
+            expected_transformed_data = pix_in_mem.data;
```

Comment:
rather than `transformed_data` and `expected_transformed_data` these are `in_memory` and `file_backed` data